### DISCUSSION TITLE
Is changing work_mem limited in some way?

### DISCUSSION THREAD

**Discussion by Zloka** [supabase/supabase#5168] [Questions] [answered]
A query of mine needs to organize a large amount of rows, and the default 4Mb `work_mem` value is not enough, so it starts writing to temporary files on disk.

A quick bandaid solution would be to increase `work_mem`, but are there any limitations to this? For example, could I risk running into out of memory errors, or see an increase in billing?

**Comment by steve-chavez** [comment, marked as answer]
> A quick bandaid solution would be to increase work_mem, but are there any limitations to this? For example, could I risk running into out of memory errors, 

Yes, the free tier db instance comes with 1GB of memory, if you surpass that limit then you'll likely hit into OOM errors.

> or see an increase in billing?

No, though if you hit OOM, you'd need to upgrade your instance memory to clear the errors.

**Reply by Zloka** [reply]
Thank you for clearing this up for me! 🙂

**Reply by Zloka** [reply]
Didn't think to ask since 1Gb will likely be more than enough, but we actually use the "Pay as you go"-plan, are the limits higher?
